### PR TITLE
Truth table style unit tests for `or` conditional

### PR DESCRIPTION
**Pull request by alallier**

### CODE REVIEW

**Comment by alallier** [comment]
Not sure if this is a good pattern or not. Let me know what you think. If it is it could be applied to other tests.

File: test/templates/conditionals/orRightSide.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{!
+  should evaluate <if something or notDefined> as true
+!}
+
+    {! true !}   {! false !}
```
Comment (alallier) [inline test/templates/conditionals/orRightSide.html]: Need to swap these comments

**Comment by alallier** [comment]
Also I beg the question should we test the truth table for conditionals? Might be useful in `and` conditionals

| var 1 | var 2 |
|-------|-------|
| true  | true  |
| true  | false |
| false | true  |
| false | false |

File: test/templates/conditionals/orRightSide.html (hunk 1 above)
Comment (kethinov) [inline test/templates/conditionals/orRightSide.html]: why not do it in your diff...?

Comment (alallier) [inline test/templates/conditionals/orRightSide.html]: This was just a reminder to go back and do it

**Comment by kethinov** [comment]
truth table is a go. i'll merge this when those changes are in

**Comment by alallier** [comment]
Updated

**Comment by kethinov** [comment]
this test only failed because of typos: https://github.com/kethinov/teddy/commit/24cb773e1149cfe8975cb3da6b738ef43ea7dcdc

**Comment by alallier** [comment]
So you can't have an or conditional that is `false false` inside of an if statement? We may want to document that clearly if that's the case

**Comment by kethinov** [comment]
The original if statement was doing what it was supposed to do.

```html
    {! false !}                  {! false !}
 <if something='Nope' or somethingMore='Nope'>
   <p>or: fasle false</p>
 </if> 
```

You passed two false conditions to an `or`. Thus the contents of the `<if>` won't render because the entire expression returned false.

**Pull request merged**